upstairs/src/client.rs

```diff
@@ -783,6 +783,14 @@ impl DownstairsClient {
                     mode: ConnectionMode::New,
                 };
             }
+            DsState::Stopping(reason) => {
```

Comment:
I think we want to limit this case to just this one reason:
```
DsState::Stopping(ClientStopReason::Replacing)
```
I've not fully determined that any other stopping state is okay to move forward.

Reply:
Oh, and thanks for submitting the PR!

Reply:
Good point. I made this change and also pulled in @mkeeter's suggested integration test in [this comment](https://github.com/oxidecomputer/crucible/issues/1716#issuecomment-2873255432).

I leveraged the `integration-test` feature to add the 100ms delay.

Thanks for the review!

Reply:
Oh and I added some comments as suggested. I did a bit of digging to try and understand what's going on there. Do those comments make sense?